lib/ConditionUtils/core/Utils.py

```diff
@@ -76,21 +79,66 @@ def file_to_condition_set(self, params):
         })[0]
         return {"condition_set_ref": "%s/%s/%s" % (info[6], info[0], info[4])}
 
+    def _conditionset_data_to_df(self, data):
+        """
+        Converts a compound set object data to a dataframe
+        """
+
+        factors = pd.DataFrame(data['factors'])
+        factors.rename(columns=lambda x: x.replace("ont", "ontology").capitalize().replace("_", " "))
+        conditions = pd.DataFrame(data['conditions'])
+        cs_df = factors.join(conditions)
+
+        return cs_df
+
+    def _clusterset_data_to_df(self, data):
+        """
+        Converts a cluster set object data to a dataframe
+        """
+
+        original_matrix_ref = data.get('original_data')
+        data_matrix = self.gen_api.fetch_data({'obj_ref': original_matrix_ref}).get('data_matrix')
+
+        data_df = pd.read_json(data_matrix)
+
+        clusters = data.get('clusters')
+
+        data_index = data_df.index.tolist()
+        cluster_name = [None] * len(data_index)
+
+        cluster_id = 0
+        for cluster in clusters:
+            gene_ids = cluster.get('id_to_data_position').keys()
+            gene_idx = [data_index.index(gene_id) for gene_id in gene_ids]
```

Comment:
I think https://pandas.pydata.org/pandas-docs/stable/generated/pandas.Index.get_loc.html would do the same thing more efficiently

Reply:
👍